helm-charts/splunk-otel-collector/templates/clusterRole.yaml

```diff
@@ -17,6 +17,7 @@ rules:
 - apiGroups:
   - ""
   resources:
+  - configmaps
```

Comment:
Can you please elaborate how adding this provides support of `kubernetes-events`?

Reply:
we can observe that leader election doesn’t have permission to write into configmaps. I am attaching an error message for that.
<img width="1773" alt="Screenshot 2021-09-01 at 9 24 19 AM" src="https://user-images.githubusercontent.com/88535006/131611032-4df92a4c-ea61-48b7-b210-5fb7fc100c09.png">
So instead of granting cluster-admin privileges to the service account, we can add write permissions to configmap for leader election.

Reply:
Ok got it. We should add it to k8sClusterReceiver deployment, not to the agent. It may work there without these changes since it's 1-replica deployment. Then let's add an option `k8sEventsEnabled: false` in values.yaml and put everything that is needed events that including the receiver itself under this flag.

Reply:
Hi @dmitryax 
I can see that this error comes in both deployments and daemonset of `otelCollector`, `otelAgent`, and `otelK8sClusterReceiver`. so I have added environment variables in both developments and daemonset. and also as you suggested I have added the `k8sEventsEnabled: false` flag in values.yaml file you can verify it and let me know if more changes are required.

Reply:
We should enable events collection on `otelK8sClusterReceiver`, so the env vars are needed only for that deployment

Reply:
We should enable events collection on `otelK8sClusterReceiver`, so the env vars are needed only for that deployment

Reply:
Hi @dmitryax 
As you suggested I have added environment variables in the only deployment of `otelK8sClusterReceiver` if fag `k8sEventsEnabled:` has ` true` value. but if possible then can you look into Salesforce Properties `Case: 00017478` comment `5` which is given in [OTL-951](https://signalfuse.atlassian.net/browse/OTL-951), in that they recommend having smartagent/Kubernetes-events under the `otelAgent` daemonset. but we are putting it in only `otelK8sClusterReceiver`. so it will not create problems in the future right?

Reply:
I'm not sure what is the comment you refer to. I'm pretty confident that we should add `smartagent/kubernetes-events` monitor to `k8sClusterReceiver` deployment. But we can ask @jrcamp (as creator of the task) if he agrees.

Reply:
I have asked Jay Campa this. He agrees with you so we have to add support for only `otelK8sClusterReceiver`.